lfs/client.go

```diff
@@ -125,8 +126,15 @@ func Download(oid string) (io.ReadCloser, int64, error) {
 		return nil, 0, err
 	}
 	LogTransfer("lfs.data.download", res)
+	
+	buf, _ := ioutil.ReadAll(res.Body)
+	body := myReader{bytes.NewBuffer(buf)}	
```

Comment:
Each `*http.Client` is designed to be goroutine-safe in an application. Its [Transport](https://golang.org/pkg/net/http/#Transport) has some properties for configuring its keepalive connections. Other than `CancelRequest()` and `CloseIdleConnections()`, it doesn't seem to give much visibility into how it's managing the connections. I wonder if we can just set `MaxIdleConnsPerHost` to 1? That would hurt concurrency for NTLM users, but should force a single connection per host.